test/ng/compileSpec.js

```diff
@@ -12111,6 +12142,890 @@ describe('$compile', function() {
+      }));
+    });
+
+    describe('form[action]', function() {
+      it('should pass through action property for the same domain', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = 'different_page';
+        $rootScope.$apply();
+        expect(element.prop('action')).toMatch(/\/different_page$/);
+      }));
+
+      it('should clear out action property for a different domain', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = 'http://a.different.domain.example.com';
+        expect(function() { $rootScope.$apply(); }).toThrowMinErr(
+            '$sce', 'insecurl', 'Blocked loading resource from url not allowed by $sceDelegate policy.' +
+            '  URL: http://a.different.domain.example.com');
+      }));
+
+      it('should clear out JS action property', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = 'javascript:alert(1);';
+        expect(function() { $rootScope.$apply(); }).toThrowMinErr(
+            '$sce', 'insecurl', 'Blocked loading resource from url not allowed by $sceDelegate policy.' +
+            '  URL: javascript:alert(1);');
+      }));
+
+      it('should clear out non-resource_url action property', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = $sce.trustAsUrl('javascript:doTrustedStuff()');
+        expect(function() { $rootScope.$apply(); }).toThrowMinErr(
+            '$sce', 'insecurl', 'Blocked loading resource from url not allowed by $sceDelegate policy.' +
+            '  URL: javascript:doTrustedStuff()');
+      }));
+
+
+      it('should pass through $sce.trustAs() values in action property', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = $sce.trustAsResourceUrl('javascript:doTrustedStuff()');
+        $rootScope.$apply();
+
+        expect(element.prop('action')).toEqual('javascript:doTrustedStuff()');
+      }));
+    });
+
+    describe('link[href]', function() {
+      it('should reject invalid RESOURCE_URLs', inject(function($compile, $rootScope) {
+        element = $compile('<link ng-prop-href="testUrl" rel="stylesheet" />')($rootScope);
+        $rootScope.testUrl = 'https://evil.example.org/css.css';
+        expect(function() { $rootScope.$apply(); }).toThrowMinErr(
+            '$sce', 'insecurl', 'Blocked loading resource from url not allowed by $sceDelegate policy.' +
+            '  URL: https://evil.example.org/css.css');
+      }));
+
+      it('should accept valid RESOURCE_URLs', inject(function($compile, $rootScope, $sce) {
```

Comment:
NIT: It is not about "valid"/"invalid". It's more about "trusted"/"untrusted".

Reply:
Attr version... (I definitely agree with your comment, but should it differ?)

Reply:
Generally, when I find something wrong I try to fix it (or at least not replicate it 😛).
But it is not critical or anything.

Reply:
Sounds good. Shall I also fix the attr version?

Reply:
Actually... when you say "trusted" does that mean an instanced of the `$sce.trustAs*()` object? Because this tests both that object as well as other "valid" things like relative paths. But that "valid" word can also be swapped for "trusted"... preference?

Reply:
By "trusted" I mean anything that is considered safe (either by being wrapped in `$sce.trustAs()` or being included in a whitelist or having some other property (e.g. being a relative URL) that makes it safe in a particular context).